contracts/tokenwrapper/src/utils.rs

```diff
@@ -0,0 +1,69 @@
+use cosmwasm_std::{Addr, Decimal, DepsMut, Fraction, MessageInfo, StdError, Uint128};
+use cw20::BalanceResponse;
+use cw20_base::contract::query_balance;
+use protocol_cosmwasm::error::ContractError;
+
+use crate::state::{CONFIG, TOKENS, TOTAL_SUPPLY};
+
+// Check if the cw20 token address is valid in "TOKENS".
+pub fn is_valid_address(deps: DepsMut, token_address: Addr) -> bool {
+    TOKENS.load(deps.storage, token_address).unwrap_or(false)
+}
+
+// Check if the "wrap_amount" is valid.
+pub fn is_valid_wrap_amount(deps: DepsMut, amount: Uint128) -> bool {
+    let total_supply = TOTAL_SUPPLY.load(deps.storage).unwrap().issued;
+    let config = CONFIG.load(deps.storage).unwrap();
+    amount + total_supply <= config.wrapping_limit
```

Comment:
Are there saturating (safe) methods for addition we should consider using?

Reply:
Yes, there is saturating addition for `Uint128`.
Will check & fix not only this one, but also other operations.

Reply:
Done https://github.com/webb-tools/protocol-cosmwasm/pull/30/commits/1f8f2b4d9a6e24b4c0ee5b9075756055faebe54a